### PR TITLE
Add scripts for KubeRay and Ray Cluster

### PR DESCRIPTION
**Pull request by xwu99**
## Description

Ray-based implementations such as dataprep, text-generation and finetuning will be based on deploying a Ray Cluster. 
Adding installation scripts for KubeRay (K8S operator) and starting an autoscaling Ray Cluster for K8S.

## Issues

N/A

## Type of change

List the type of change like below. Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would break existing design and interface)

## Dependencies

[KubeRay](https://github.com/ray-project/kuberay)

## Tests

CI tests added.

### CODE REVIEW

**Comment by xwu99** [comment]
@daisy-ycguo we would like to add scripts to install kuberay and start Ray cluster which are essentials to support dataprep, text-generation and finetuning on Ray. Could you help to review this first and kick off the CI,  thanks! 

After this, we will be working on finetuning RFC and code later.

**Review by xuechendi — APPROVED** [review]
LGTM

File: scripts/ray/ray-cluster-autoscaler.yaml [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+            requests:
+              cpu: "1"
+              memory: "2G"
+          volumeMounts:
+            - name: data-volumne
+              mountPath: /data
+        volumes:
+          - name: data-volumne
+            hostPath:
+              path: /mnt
```
Comment (xuechendi) [inline scripts/ray/ray-cluster-autoscaler.yaml:88]: is `/mnt` on localhost?
can we also create a template file to expose volumes, resources 'cpu', 'memory' for user to provide their configuration?

File: .github/workflows/ray-e2e.yaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  workflow_dispatch:
+
+# If there is a new commit, the previous jobs will be canceled
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}-xeon
+  cancel-in-progress: true
+
+jobs:
+  ray-validate:
+    runs-on: kind-xeon
```
Comment (daisy-ycguo) [inline .github/workflows/ray-e2e.yaml:22]: please change it to `xeon`. Thanks.

Comment (xwu99) [reply .github/workflows/ray-e2e.yaml]: updated.

**Review by daisy-ycguo — APPROVED** [review]
lgtm

**Comment by daisy-ycguo** [comment]
> @daisy-ycguo we would like to add scripts to install kuberay and start Ray cluster which are essentials to support dataprep, text-generation and finetuning on Ray. Could you help to review this first and kick off the CI, thanks!
> 
> After this, we will be working on finetuning RFC and Ray code later.

After discussing with @xwu99 In order to support Ray workloads, CI infrastructure will install KubeRay directly on top of K8s cluster as a infrastructure capability. We will not add any CI workflows in GenAIInfra till GenAIComps and GenAIExamples are ready.

**Review by lianhao — APPROVED** [review]

**Review by daisy-ycguo — APPROVED** [review]
lgtm

**Pull request merged**